llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -164,4 +215,32 @@ def _save_checkpoint(self, state: State, logger: Logger):
                             overwrite=self.overwrite,
                         )
 
-        dist.barrier()
+                elapsed_duration = state.get_elapsed_duration()
+                if self.log_to_mlflow and elapsed_duration is not None and elapsed_duration >= 1.0:
+                    components = {'model': new_model_instance}
+                    if original_tokenizer is not None:
+                        components['tokenizer'] = original_tokenizer
+
+                    log.debug('Logging Hugging Face model to MLFlow')
+                    registered_model_name = f'{state.run_name}_{os.path.basename(save_dir)}'
```

Comment:
Looks like this will produce a new registered model on each run (?) since `save_dir` includes a timestamp. I wonder if there's a better way we can group together the model versions produced by these runs - it's not recommended to create a new registered model per run since this complicates deployment workflows (you can't leverage features like model aliases to manage which model version out of a group of models is deployed for serving/inference).

Is there some user-provided name we can reuse as the registered model, or require as input to enable model registration? 

For tracking model metadata like the run timestamp etc of the model, I'd recommend using [set_model_version_tag ](https://mlflow.org/docs/latest/python_api/mlflow.client.html?highlight=set_model_version_tag#mlflow.client.MlflowClient.set_model_version_tag)(basically setting metadata tags on the created model version) via something like:

```
model_version =    mlflow_logger.register_model(
                            model_uri=local_save_path,
                            name=registered_model_name,
                            await_registration_for=3600,
)

# We should use the MlflowClient instance associated with the logger, but including this as an example
MlflowClient().set_model_version_tag(name=registered_model_name, version=model_version.version, key=..., value=...)
```

Tags are also searchable etc (although not yet in UC...)

Reply:
hmm, I can remove the timestamp info from the run and then it would just be the run name. This would still be different for each training run. We could force the user to provide a model name to register under, but I'd be a bit worried about this getting confusing (the less arguments the better, and users tend to forget to change stuff between runs, and so might end up registering things under the same model that they didn't mean to). Maybe we can start here by just using the run name and see what the usage feedback is? I don't think I have a good understanding of how users' workflows are going to look around deploying to databricks yet.

Reply:
ok, i think i'll just make the registered model name a required arg

Reply:
Lol, sorry for the slow response - yes I think that would make sense, I was gonna post this:

> Ah thanks, got it, I can share some context on model deployment workflows in Databricks (for just ML models in general - my knowledge isn't LLM-specific but this stuff seems somewhat general). In general the pattern is to use a single registered model per business problem, e.g. group model versions for `revenue_forecasting` or `customer_support_bot` under the same registered model.

Reply:
In the end, I'd hope customers can automate production model training to catch / prevent issues like training a model for business problem A and registering it to a registered model for business problem B

Reply:
ah ok, that is all helpful context, thanks!